src/Aquifer.Data/Entities/VersificationMappingEntity.cs

```diff
@@ -0,0 +1,21 @@
+using Aquifer.Data.EventHandlers;
+
+namespace Aquifer.Data.Entities;
+
+public class VersificationMappingEntity : IHasUpdatedTimestamp
+{
+    public int Id { get; private set; }
```

Comment:
I kept thinking about making a composite key with this, but I started thinking it may have too many pieces to it to keep it unique, so I thought maybe an Id was best... not sure if that was the right choice though

Reply:
Yeah, I agree that this one has a lot of pieces to it to make a unique item. @NateMerritt Do you have any thoughts on this? The combination of `BibleId`, `BibleVerseId`, and `VerseIdPart` should be unique. Is there any issue putting a unique index on those three with one of them being nullable char, or a triple primary key?

Reply:
I think it would also be the `BaseVerseId` and maybe even `BaseVerseIdPart` as well, because of the many to many relationships.

Reply:
The PK column value cannot be `null`.  I haven't tested that out with multi-column PKs but I'm expecting it to still apply.  A unique constraint should work, though.

I've generally avoided multi-column PKs for a few reasons:
1. Insert performance: An auto-incrementing `int` or `bigint` PK column performs better on inserts because multi-column inserts can be anywhere in the primary index but for auto-incrementing PKs any new row is always added at the end of the primary index.  Inserting into the middle of the index (B-tree) eventually requires rebalancing as part of the insert operation which can take many seconds for larger tables (I've seen it take over 30 seconds for a table with multiple millions of rows).  That said, this data should never have inserts/updates, only possible re-loads, so that's not really an issue.
2. FKs to the PK require multiple columns which makes queries more complex because all columns must be in the WHERE clause.
3. API usage: Generally you want a single unique ID for Get by ID requests when a resource is exposed over an API.

I do think they make sense in some usages, though.